### PR TITLE
Support viewer to download `gardenlogin` kubeconfig by fetching cluster CA via `ConfigMap`

### LINKED ISSUE #1709: Viewers should be able to download `gardenlogin` kubeconfig
**What would you like to be added**:
`gardenlogin` `v0.5` or higher supports the `shoots/viewerkubeconfig` subresource. As a result, users with the `Project` `viewer` role can now access the Shoot cluster with read-only permission. In addition, the cluster CA can now be fetched via `ConfigMap` (https://github.com/gardener/gardener/pull/9123).
Therefore, it should be possible to download the `gardenlogin` kubeconfig in the dashboard with only viewer permissions.

**Why is this needed**:

### LINKED ISSUE #1710: Viewers should be able to copy the `Target Cluster` command for a `Shoot`
**What would you like to be added**:
`gardenlogin` `v0.5` or higher supports the `shoots/viewerkubeconfig` subresource. As a result, users with the `Project` `viewer` role can now access the Shoot cluster with read-only permission. 
In addition, the cluster CA can now be fetched via `ConfigMap` (https://github.com/gardener/gardener/pull/9123). This allows the generation of the `gardenlogin` kubeconfig via `gardenctl` with only `viewer` permissions.

Therefore, it should be possible to copy the `Target Cluster` command for a `Shoot` cluster as `Project` `viewer`.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by petersutter**
**What this PR does / why we need it**:
This PR enables users with the `Project` `viewer` role to download the `gardenlogin` kubeconfig by fetching the cluster CA via the `<shoot-name>.ca-cluster` `ConfigMap`.

To build the `gardenlogin` kubeconfig when targeting a `Shoot` cluster, the Dashboard needs to read the cluster CA. In earlier versions, this was obtained from a `.ca-cluster` `Secret`. However, starting from Gardener `v1.89`, the cluster CA is also reconciled as a `ConfigMap`. This update allows users with the `viewer` role to read the `ConfigMap`, thereby facilitating the generation of the `gardenlogin` kubeconfig.

Side note:
`gardenlogin` `v0.5` or higher is designed to fetch credentials from two subresources: `shoots/adminkubeconfig` and `shoots/viewerkubeconfig`. By default, it retrieves credentials from the `shoots/adminkubeconfig` subresource, providing full administrative access. Alternatively, it can fetch credentials from the `shoots/viewerkubeconfig` subresource for read-only access.

The plugin automatically adjusts the level of access for the fetched credentials. It first attempts to fetch admin-level credentials and, if unsuccessful, falls back to viewer-level credentials. This feature makes the `gardenlogin` kubeconfig versatile, as it can be used by both project admins and viewers without the need to specify the access level.

**Which issue(s) this PR fixes**:
Fixes #1709, fixes #1710

**Special notes for your reviewer**:
ref https://github.com/gardener/gardener/issues/9091

**Release note**:

```bugfix user
Users with the `Project` `viewer` role are now able to to download the `gardenlogin` kubeconfig by fetching the cluster CA via `ConfigMap`. This feature is supported with Gardener `v1.89` and requires `gardenlogin` `v0.5` or higher.
```

### CODE REVIEW

File: backend/lib/services/shoots.js [hunk 1]
```diff
@@ -416,14 +422,23 @@ async function getKubeconfigGardenlogin (client, shoot) {
   const { namespace, name } = shoot.metadata
 
   const [
-    ca,
-    gardenClusterIdentity
-  ] = await Promise.all([
-    client.core.secrets.get(namespace, `${name}.ca-cluster`),
+    { value: configmap, reason: configmapError },
```
Comment (holgerkoser) [inline backend/lib/services/shoots.js:425]: Define the function `getGardenClusterCaData` here, export this function and use it in the terminals code. 
```js
async function getGardenClusterCaData (client, { namespace, name }) {
  try {
    const configmap = await client.core.configmaps.get(namespace, `${name}.ca-cluster`)
    return encodeBase64(configmap.data?.['ca.crt'])
  } catch (err) {
    // TODO(petersutter): Remove this fallback of reading the `<shoot-name>.ca-cluster` Secret when Gardener no longer reconciles it, presumably with Gardener v1.97.
    if (isHttpError(err, 404)) {
      const caCluster = await client.core.secrets.get(namespace, `${name}.ca-cluster`)
      return caCluster.data['ca.crt']
    } else {
      throw err
    }
  }
}
exports.getGardenClusterCaData = getGardenClusterCaData
```

Comment (petersutter) [reply backend/lib/services/shoots.js]: ✅

**Review by grolu — APPROVED** [review]
/lgtm

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Pull request merged**